x-pack/plugins/code/public/components/editor/editor.tsx

```diff
@@ -55,6 +55,24 @@ export class EditorComponent extends React.Component<IProps> {
     super(props, context);
   }
 
+  updateGutterClickHandler = (
+    repoUri: string,
+    revision: string,
+    path: string,
+    queryString: string
+  ) => {
+    if (this.monaco && this.editor) {
+      this.editor!.onMouseDown((e: editorInterfaces.IEditorMouseEvent) => {
```

Comment:
Why this event handler fixes the first issue?

Reply:
see https://github.com/elastic/kibana/pull/38748/files#diff-d34b40ad2e09a0db78dda59564fc5021R108, if `query string` changes, it would `loadText`, but we only want to update gutter click handler when `query string` changes.

Reply:
see https://github.com/elastic/kibana/pull/38748/files#diff-d34b40ad2e09a0db78dda59564fc5021R108, if `query string` changes, it would `loadText`, but we only want to update gutter click handler when `query string` changes.

Reply:
Ok, please check if `onMouseDown` deletes the old handler before adding a new one. I'm worried multiple handles were adding here. Maybe you should remove the handler in `componentWillUnmount`.